pypsa/components.py

```diff
@@ -1167,13 +1167,28 @@ def consistency_check(self):
                 logger.warning("The following %s have buses which are not defined:\n%s",
                                c.list_name, missing)
 
+        # check for unknown buses
         for c in self.iterate_components(self.branch_components):
             for attr in ["bus0","bus1"]:
                 missing = c.df.index[~c.df[attr].isin(self.buses.index)]
                 if len(missing) > 0:
                     logger.warning("The following %s have %s which are not defined:\n%s",
                                    c.list_name, attr, missing)
 
+        # check for disconnected buses
+        connected_buses = set()
+        for c in self.iterate_components(self.branch_components):
+            for attr in ["bus0","bus1"]:
+                connected_buses.update(c.df[attr])
+        for c in self.iterate_components(self.one_port_components):
+            for attr in ["bus"]:
+                connected_buses.update(c.df[attr])
+
+        disconnected_buses = list(set(self.buses.index) - connected_buses)
+        if len(disconnected_buses) > 0:
```

Comment:
```suggestion
        disconnected_buses = set(self.buses.index) - connected_buses
        if disconnected_buses:
```
requires a bit less computation.